testcontainers/localstack.py

```diff
@@ -0,0 +1,58 @@
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+from testcontainers.core.waiting_utils import wait_for_logs
+from testcontainers.core.container import DockerContainer
+
+
+class LocalStackContainer(DockerContainer):
+    """
+    Localstack container.
+
+    Example
+    -------
+    ::
+        localstack = LocalStackContainer(image="localstack/localstack:0.11.3")
+        localstack.with_services("dynamodb", "lambda")
+        localstack.start()
+        dynamo_endpoint = localstack.get_service_endpoint()
+        dynamo_client = boto3.client("dynamodb", endpoint_url=dynamo_endpoint)
+        scan_result = dynamo_client.scan(TableName='foo')
+        # Do something with the scan result
+    """
+    EDGE_PORT = 4566
+    IMAGE = 'localstack/localstack:0.11.3'
+
+    def __init__(self, image=IMAGE):
+        super(LocalStackContainer, self).__init__(image)
+        self.with_exposed_ports(LocalStackContainer.EDGE_PORT)
+        self.with_volume_mapping('/var/run/docker.sock', '/var/run/docker.sock')
```

Comment:
Is this line necessary? Does localstack spin up new containers in some circumstances?

Reply:
For lambdas it needs to spin up containers yes. Most of the time I use this so I added it.
I can remove it or create a function doing this, WDYT ?

Reply:
Ah, interesting. Would it be an idea to only set the volume mapping if the `lambda` service is requested? In some CI environments (such as ours) it would not be desirable for additional containers to be created, as they might not get cleaned up afterwards

Reply:
I am not sure that only lambda needs this to be honest. Also some other ENV vars can change this behavior and make the lambda run locally instead of new containers.
So for now I think it's better if we let the users chose when to put this volume mapping according to what's span.
I will remove it.